packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation.js

```diff
@@ -57,71 +55,107 @@ export const usePaymentInformation = props => {
-             *
-             * Do not call if paymentNonceDetails is not in cache, because we
-             * need it to render the summary. There is no way to save it on
-             * the remote server, hence we save it in local cache.
-             */
-            const selectedPaymentMethodOnCart =
-                paymentDetailsData &&
-                paymentDetailsData.cart.selectedPaymentMethod
-                    ? paymentDetailsData.cart.selectedPaymentMethod.code
-                    : null;
-            const paymentNonceDetails = paymentDetailsData
-                ? paymentDetailsData.cart.paymentNonce
-                : null;
-
-            const hasData = !!(
-                selectedPaymentMethodOnCart && paymentNonceDetails
-            );
-
-            if (hasData) {
-                if (onSave) {
-                    onSave();
-                }
-            }
-
-            setHasData(hasData);
-            setSkipQueryCall(true);
-        },
-        [onSave]
-    );
-
     /**
      * Queries
      */
+    const {
+        data: paymentInformationData,
+        loading: paymentInformationLoading
+    } = useQuery(getPaymentInformation, {
+        variables: { cartId }
+    });
+    const [
+        setPaymentMethod,
+        { loading: setPaymentMethodLoading }
+    ] = useMutation(setPaymentMethodMutation);
+    /**
+     * Effects
+     */
 
-    useQuery(getPaymentDetailsQuery, {
-        variables: { cartId },
-        skip: skipQueryCall,
-        onCompleted: onPaymentDetailsQueryCompleted
+    const availablePaymentMethods = paymentInformationData
+        ? paymentInformationData.cart.available_payment_methods
+        : [];
+
+    const selectedPaymentMethod =
+        (paymentInformationData &&
+            paymentInformationData.cart.selected_payment_method.code) ||
+        null;
+
+    // Whenever available methods change we should reset to the editing view
+    // so that a user can see the newly available methods. This could occur
+    // if a user causes their cart total to become $0. Additionally, the
+    // current pattern requires that the method components themselves
+    // indicate their "done" state so we must leave it to them to revert
+    // this effect downstream.
+    useEffect(() => {
+        setDoneEditing(false);
+    }, [availablePaymentMethods]);
+
+    // Along the lines of the above effect, since we don't have a "free method"
+    // component we check if free is available and set it as the selected method
+    // directly. What results is a potential flash of selectable payment methods
+    // and then the "free" summary.
+    useEffect(() => {
+        const setFreeIfAvailable = async () => {
+            const freeIsAvailable = !!availablePaymentMethods.find(
+                ({ code }) => code === 'free'
+            );
+            if (freeIsAvailable) {
+                if (selectedPaymentMethod !== 'free') {
+                    await setPaymentMethod({
```

Comment:
I feel we can move this logic into the Free component's talon and handle it as any other payment method would. For instance: https://github.com/magento/pwa-studio/pull/2409/files#diff-34f11d4c09662e347b956c693ec2ca43

Reply:
I'll leave my response in the other thread.